pkg/server/router.go

```diff
@@ -57,6 +57,7 @@ func router(ctx context.Context, config *Config, cfg *cmds.Server) http.Handler
 	serverAuthed := mux.NewRouter()
 	serverAuthed.Use(authMiddleware(serverConfig, version.Program+":server"))
 	serverAuthed.NotFoundHandler = nodeAuthed
+	serverAuthed.Path(prefix + "/configServer").Handler(configToCompareHandler(serverConfig, cfg))
```

Comment:
Do we really need a separate route and handler for this? You can already get pretty much everything you need from the existing config endpoint:
```
authed.Path(prefix + "/config").Handler(configHandler(serverConfig, cfg))
```
From there the new server can just compare the other server's config to its local config and fail if something's out of sync.

Reply:
That was my first approach. Two reasons why I did not proceed with that one:

1. `/config` is auth only for agents. Servers get an Unauthorized message. We need to add another route using `serverAuthed`.
2. While discussing with Brian, we thought that exposing the whole config creates security risks. If we expose just a hash, those security risks are removed and we can still compare the config

Reply:
The auth thing should not be an issue. Every server is also an agent; it sounds like you're making the request with the wrong username - use `node` for agent, `server` for server, with the same token for either as the password.

As far as I know, all of the fields you're trying to compare are already exposed via /config. The only ones not exposed  via /config have a `json:"-"` tag on them in the config.Control struct: https://github.com/k3s-io/k3s/blob/37fcb61f5ea3c843d0ac89ac82c72f3c5f0d77f5/pkg/daemons/config/types.go#L104